cmake/flake8.cmake

```diff
@@ -0,0 +1,49 @@
+#
+# Setup running flake8 on python scripts to enforce PEP8 
+# NOTE: Currently skips the check if flake8 is not installed. PRs editing python scripts are rare
+#       so don't want to add a hard dependency on flake8 to all builds.
+#
+find_program(flake8_BIN NAMES flake8)
+if(NOT flake8_BIN)
+    message(WARNING "Could not find 'flake8' to check python scripts. Please install flake8 using pip.")
```

Comment:
How about changing this to FATAL_ERROR to make sure flake8 is executed locally before user submits a long CI job, wasting lots of resource on Azure

Reply:
As per the reason in the comment above I don't want to add a hard dependency for all builds on flake8 being installed. Forcing someone trying to build ORT from source to install flake8 seems unnecessarily burdensome as it's unrelated to the core product. If someone is editing python files yes the first CI build will fail but presumably they will install flake8 and their local builds from then on will have the check. A one-off cost to that developer seems a better balance.